sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -315,6 +317,11 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 	}}
 	events := component.NewEvent(sensorEvents...)
 	events.AddSensorEvent(secretToSensorEvent(action, protoSecret))
+
+	// When adding new docker config secrets we need to reprocess every deployment in this cluster.
+	// This is because the field `NotPullable` could be updated and hence new image scan results will appear.
+	events.AddDeploymentReference(resolver.ResolveAllDeployments(), central.ResourceAction_UPDATE_RESOURCE, false)
```

Comment:
Should we guard this behind the re-sync envVar? It is ignored by the `OutputQueue` if re-sync is enabled, so technically it doesn't matter if it's guarded or not but, since is something we only need to do in the re-sync-less path, it might be clearer if it is.

Reply:
Done